endpoints/openrtb2/auction_test.go

```diff
@@ -2912,6 +2912,108 @@ func TestValidateNativeAssetData(t *testing.T) {
 	}
 }
 
+func TestFPDHoldAuction(t *testing.T) {
+
+	deps := &endpointDeps{
+		fakeUUIDGenerator{},
+		&warningsCheckExchange{},
+		newParamsValidator(t),
+		&mockStoredReqFetcher{},
+		empty_fetcher.EmptyFetcher{},
+		empty_fetcher.EmptyFetcher{},
+		nil,
+		&metricsConfig.DummyMetricsEngine{},
+		analyticsConf.NewPBSAnalytics(&config.Analytics{}),
+		map[string]string{},
+		false,
+		[]byte{},
+		openrtb_ext.BuildBidderMap(),
+		nil,
+		nil,
+		hardcodedResponseIPValidator{response: true},
+	}
+
+	if specFiles, err := ioutil.ReadDir("../../firstpartydata/tests/holdauction"); err == nil {
+		for _, specFile := range specFiles {
+			fileName := "../../firstpartydata/tests/holdauction/" + specFile.Name()
```

Comment:
Why do we have two different sources of FPD tests for the auction endpoint? I see some in the `openrtb/sample-requests/first-party-data` and then these ones which are coming from the `firstpartydata` package. I'd expect the Auction endpoint level JSON tests to only be coming only from the former spot. These additional JSON tests coming from the `firstpartydata` package don't seem right to me. Also, remember that at this level we shouldn't try to test each and every usecase. These test are more meant to make sure that the downstream functions are called as expected and that integration works. Exhaustive test cases should be implemented at the downstream level.

Reply:
Agree, moved.
Brian, please re-review `TestFPDHoldAuction`. Looks like it does what you mentioned.